language-server/modules/langserver-compiler/src/main/java/org/ballerinalang/langserver/compiler/config/ExtendedClientConfigListener.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.compiler.config;
+
+import com.google.gson.JsonElement;
+
+/**
+ * Represents a ballerina client config change listener.
+ *
+ * @since 2.0.0
+ */
+@FunctionalInterface
+public interface ExtendedClientConfigListener extends BaseClientConfigListener {
```

Comment:
Can't we depend only on the `didChangeConfig`? If so we do not need to have two interfaces for `ExtendedClientConfigListener` and `ClientConfigListener`. Now the hierarchy have three interfaces.

Reply:
We do need both `didChangeConfig` and `didChangeJson` for different use cases. As discussed during the call, I'll remove the inherited interfaces for simplicity.